.github/workflows/test.yml

```diff
@@ -33,6 +38,20 @@ jobs:
       - name: Install dependencies
         if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
         run: make setup
+
+      # Build artifacts
+      - name: Event name
+        run: echo "github.event_name=${{ github.event_name }}"
+      - name: Format all the cairo files
+        run: make format
+      - name: Commit changes
+        # Don't run on pull request events. Commit only when there are changes, otherwise it continues
+        if: github.event_name != 'pull_request'
```

Comment:
can't we add the commit to the PR instead of adding it to main?

Reply:
So on `push` event, this step adds the [commit](https://github.com/sayajin-labs/kakarot/pull/315/commits/cebd1b1e31253a4b40d5bf8b7c6428af47d69dce) to the pr

Reply:
ok got it sorry, not an expert in GH action, I had "push to main" in mind

Reply:
Ah no worries, my wording was also a bit vague 😅 Upon merging, it does a `push` to main but there won't be files modified from `make format` (since it's been committed already). Then it `git commit && git push` would be skipped and just proceeds to the compile step